src/transaction.rs

```diff
@@ -45,7 +45,19 @@ pub trait Connectional {
     /// or schema mutations.
     fn with_connection<F, T>(&self, db: &str, f: F) -> QueryResult<T>
     where
-        F: FnOnce(&mut Connection) -> QueryResult<T>;
+        F: FnOnce(&mut Connection) -> QueryResult<T>,
+        Self: Sized;
+
+    fn execute_on_connection(&self, db: &str, query: Query) -> QueryResult<Option<Id>>;
```

Comment:
All these three methods from here to the end of the trait can be implemented in the trait, so you don't need to repeat yourself in the different connectors. They all call the `with_connection`, which should be the only method required in the connectors.